src/Xamarin.Android.Build.Tasks/Utilities/Files.cs

```diff
@@ -190,26 +190,26 @@ static string HashZip (string filename)
 
 		public static ZipArchive ReadZipFile (string filename)
 		{
-			return ZipFile.Open (filename, ZipArchiveMode.Read, new System.Text.UTF8Encoding (false));
+			return ZipArchive.Open (filename, FileMode.Open);
 		}
 
 		public static void ExtractAll(ZipArchive zip, string destination, Action<int, int> progressCallback = null)
 		{
 			int i = 0;
-			int total = zip.Entries.Count;
-			foreach (var entry in zip.Entries) {
-				if (entry.FullName.Contains ("/__MACOSX/") ||
-						entry.FullName.EndsWith ("/__MACOSX", StringComparison.OrdinalIgnoreCase) ||
-						entry.FullName.EndsWith ("/.DS_Store", StringComparison.OrdinalIgnoreCase))
+			int total = (int)zip.NumberOfEntries;
+			foreach (var entry in zip) {
+				if (entry.Name.Contains ("/__MACOSX/") ||
```

Comment:
libZipSharp API question: should this be `FullName` instead of `Name`? Maybe System.IO.Compression has the right idea here...